crates/rust-analyzer/src/handlers.rs

```diff
@@ -125,6 +128,19 @@ pub(crate) fn handle_memory_usage(state: &mut GlobalState, _: ()) -> Result<Stri
     Ok(out)
 }
 
+pub(crate) fn fetch_dependency_list(
+    state: GlobalStateSnapshot,
+    _params: lsp_ext::FetchDependencyListParams,
+) -> Result<lsp_ext::FetchDependencyListResult> {
+    let crates = state.analysis.fetch_crates()?;
+    Ok(FetchDependencyListResult {
+        crates: crates
+            .into_iter()
+            .map(|it| CrateInfoResult { name: it.name, version: it.version, path: it.path })
```

Comment:
Here we need to resolve the file id to its path, or url even, with the `GlobalStateSnapshot`